layers/gre_test.go

```diff
@@ -253,3 +254,138 @@ func BenchmarkEncodePacketEthernetOverGRE(b *testing.B) {
 		buf.Clear()
 	}
 }
+
+var testGREChecksum = map[uint16][]gopacket.SerializableLayer{
+	0x77ff: {
```

Comment:
Yes, the second one was from my desktop, though I changed the outer headers that don't affect the GRE checksum.

Because the checksums in the lower level packets cause them to cancel out, only the GRE header actually affect the value so there are a limited number of values it can take, basically just the protocol field affect it in these packets. Trying to capture wild packets with other options is hard.